src/pike/smb2.py

```diff
@@ -4161,16 +4161,59 @@ def _decode(self, cur):
         self.security_mode = SecurityMode(cur.decode_uint16le())
         self.dialect = Dialect(cur.decode_uint16le())
 
+class AddressFamily(core.ValueEnum):
+    IPv4 = 0x0002
+    IPv6 = 0x0017
+
+AddressFamily.import_items(globals())
+
+
+class Capability(core.FlagEnum):
+    RSS_CAPABLE = 0x00000001
+    RDMA_CAPABLE = 0x00000002
+
+Capability.import_items(globals())
 
 class QueryNetworkInterfaceInfoResponse(IoctlOutput):
     ioctl_ctl_code = FSCTL_QUERY_NETWORK_INTERFACE_INFO
 
     def __init__(self, parent):
         IoctlOutput.__init__(self, parent)
+        self.next_loc = {}
+        self.if_index = {}
+        self.capability = {}
+        self.reserved = {}
+        self.link_speed = {}
+        self.family = {}
+        self.port = {}
+        self.ip_address = {}
+        self.flow_info = {}
+        self.scope_id = {}
+        self.reserved_2 = {}
+        self.reserved_3 = {}
 
     def _decode(self, cur):
-        pass
-
+        i = 0
+        while True :
```

Comment:
This class is the correct place to loop through all of the structures in the buffer... but these structures need to be stored as **separate objects**, not indexed into several dictionaries with the same key. No other structure in pike uses this style.